### PR TITLE
Ft rate accommodation 182201730

### PR DESCRIPTION
**Pull request by mwisemarierose**
## What does this PR do?
Users should be able to rate accomodation centre 

## Description of Task to be completed?

rate accomodation centre 

## How should this be manually tested?
clone this repository
checkout to this branch ft-rate-accommodation-182201730
run npm install to install dependencies
run migrations using npm run migrate
run npm run start:dev to start the development server
add a tripRequest: '/api/v1//user/trip
approve the tripRequest : /api/v1/user/trip/approve/${requestId}
wait for 24hrs to rate the accomodation
rate accomodation: 'api/v1/rates/createRate'

## What are the relevant pivotal tracker stories?
https://www.pivotaltracker.com/n/projects/2569105

### CODE REVIEW

**Review by Placide123 — DISMISSED** [review]

**Review by alaintresor — DISMISSED** [review]

File: src/controllers/rates.controller.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      ),
+    );
+  }
+  const isItRated = await Rates.findOne({
+    where: {
+      requesterId: req.user.id,
+      accomodationId: tripRequest.accomodationId,
+    },
+  });
+  console.log(isItRated);
```
Comment (joshwambere) [inline src/controllers/rates.controller.js]: delete this console

File: src/controllers/rates.controller.js [hunk 2]
```diff
@@ -0,0 +1,81 @@
+import db from '../database/models/index.js';
+const Rates = db['rates'];
+const Accomodation = db['accomodation'];
+const tripRequests = db['tripRequest'];
+import AppError from '../utils/appError';
+import { Op } from 'sequelize';
+import catchAsync from '../utils/catchAsync.js';
+
+export const createRate = catchAsync(async (req, res, next) => {
```
Comment (joshwambere) [inline src/controllers/rates.controller.js:36]: and please add and endpoint to get all ratings for one accommodation

**Review by joshwambere — CHANGES_REQUESTED** [review]

File: src/controllers/rates.controller.js [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    return next(
+      new AppError(
+        'Sorry, accomodation does not either exist or belong to you',
+        401,
+      ),
+    );
+  }
+  if (tripRequest.status != 'approved') {
+    return next(
+      new AppError('Sorry, trip request must be approved to proceed ', 401),
```
Comment (uwambajeddy) [inline src/controllers/rates.controller.js:57]: this logic means that if I have many trip requests, all of them must be approved to rate a single accommodation even if you've been there before. And  I guess that should not be the scenario.

**Review by uwambajeddy — CHANGES_REQUESTED** [review]

**Review by uwambajeddy — CHANGES_REQUESTED** [review]
I tried to update the rates but they remained the same
Screenshot (429)

**Pull request closed without merging**